narwhals/_pandas_like/group_by.py

```diff
@@ -144,14 +148,28 @@ def agg(self: Self, *exprs: PandasLikeExpr) -> PandasLikeDataFrame:  # noqa: PLR
                         expected_old_names.append(f"{output_name}_{function_name}")
                         simple_aggs[output_name].append(function_name)
                         simple_agg_new_names.append(alias)
+                        simple_aggs_functions.add(function_name)
 
             result_aggs = []
 
             if simple_aggs:
-                result_simple_aggs = self._grouped.agg(simple_aggs)
-                result_simple_aggs.columns = [
-                    f"{a}_{b}" for a, b in result_simple_aggs.columns
-                ]
+                # Fast path for single aggregation such as `df.groupby(...).mean()`
+                if (
+                    len(simple_aggs_functions) == 1
+                    and (agg_method := simple_aggs_functions.pop()) != "size"
```

Comment:
`size` is a special case since the same "rule" below does not apply:

```py
pd.DataFrame(data).groupby("a")[["b", "c"]].size()
```
will return a nameless series (even though there are multiple columns selected)